### DISCUSSION TITLE
Using Typescript in Mod

### DISCUSSION THREAD

**Discussion by linfan68** [Moddable-OpenSource/moddable#1104] [Q&A] [answered]
I'm trying to build a Mod with typescript.
I've wrote the Host part using typescript, and it works fine. To my understanding, by including this manifest, the tsc compiler finds the type information:
```
"$(MODDABLE)/examples/manifest_typings.json"
```

However, when I include this in a Mod's manifest like this:
```json
{
  "include": [
    "$(MODDABLE)/examples/manifest_mod.json",
    "$(MODDABLE)/examples/manifest_typings.json"
  ],
  "config": {
    "version": 1.1
  },
  "modules": {
    "*": [
      "./src/*"
    ]
  },
}

```

mcrun failed with this:
```
>mcrun -d -m -p esp32/esp32s3
# tsc tsconfig.json
../../moddable/typings/commodetto/PixelsOut.ts:28:29 - error TS2304: Cannot find name 'HostBuffer'.
28       pixels: ArrayBuffer | HostBuffer,
                               ~~~~~~~~~~
../../moddable/typings/commodetto/PixelsOut.ts:59:26 - error TS2304: Cannot find name 'HostBuffer'.
59     readonly c_dispatch: HostBuffer | undefined;
                            ~~~~~~~~~~
Found 2 errors.
NMAKE : fatal error U1077: 'tsc -p C:\DK\DKJS\moddable\build\tmp\esp32\debug\rtkClient\files\tsconfig.json' : return code '0x2'
Stop.
```

So I'm wondering if this is supported?

**Comment by phoddie** [comment, marked as answer]
Thank you for the report! Yes, TypeScript is supported for mods.

You are correct that it is necessary to include both `manifest_mod.json` and `manifest_typings.json` to use TypeScript in a mod. Unfortunately, the default `tsconfig` is missing from `manifest_typings.json` which prevents it from working in some cases. If you add these lines to `manifest_types.json` or your mod's manifest.json, it works like in a host:

```json
	"typescript": {
		"tsconfig": {
			"compilerOptions": {
				"types": [
					"$(MODDABLE)/xs/includes/xs",
					"$(MODDABLE)/typings/global"
				]
			}
		}
	}
```

We'll get that `manifest_mod.json` updated shortly.
